tests/smoke/alp_cholesky.cpp

```diff
@@ -19,56 +19,117 @@
 #include <sstream>
 #include <vector>
 
+#ifdef _COMPLEX
+#include <complex>
+#include <cmath>
+#include <iomanip>
+#endif
+
 #include <alp.hpp>
+#include <graphblas/utils/iscomplex.hpp> // use from grb
 #include <alp/algorithms/cholesky.hpp>
 #include <alp/utils/parser/MatrixFileReader.hpp>
 #include "../utils/print_alp_containers.hpp"
 
 using namespace alp;
 
-typedef double ScalarType;
-constexpr ScalarType tol = 1.e-10;
+using BaseScalarType = double;
+
+#ifdef _COMPLEX
+using ScalarType = std::complex< BaseScalarType >;
+//not fully implemented structures
+using HermitianOrSymmetric = structures::Hermitian;
+#else
+using ScalarType = BaseScalarType;
+//fully implemented structures
+using HermitianOrSymmetric = structures::Symmetric;
+#endif
+constexpr BaseScalarType tol = 1.e-10;
 constexpr size_t RNDSEED = 1;
 
+template< typename T >
+T random_value();
+
+template<>
+BaseScalarType random_value< BaseScalarType >() {
+	return static_cast< BaseScalarType >( rand() ) / RAND_MAX;
+}
+
+template<>
+std::complex< BaseScalarType > random_value< std::complex< BaseScalarType > >() {
+	const BaseScalarType re = random_value< BaseScalarType >();
+	const BaseScalarType im = random_value< BaseScalarType >();
+	return std::complex< BaseScalarType >( re, im );
+}
+
+
 struct inpdata {
 	std::string fname="";
 	size_t N=0;
 };
 
-//** gnerate upper/lower triangular part of a SPD matrix */
-template< typename T >
-void generate_spd_matrix_full( size_t N, std::vector<T> &data ) {
+
+//** generate full storage Symmetric or Hermitian positive definite matrix for in-place tests */
+template<
+	typename T
+>
+std::vector< T >  generate_symmherm_pos_def_mat_data_full(
+	size_t N
+) {
+	std::vector< T > data( N * N );
+	std::fill(data.begin(), data.end(), static_cast< T >( 0 ) );
 	for( size_t i = 0; i < N; ++i ) {
-		for( size_t j = 0; j < N; ++j ) {
-			size_t k = i * N + j;
-			if( i <= j ) {
-				data[ k ] = static_cast< T >( std::rand() ) / static_cast< T >( RAND_MAX );
-			}
+		for( size_t j = i; j < N; ++j ) {
+			data[ i * N + j ] = random_value< T >();
+			data[ j * N + i ] += grb::utils::is_complex< T >::conjugate( data[ i * N + j ] );
 			if( i == j ) {
-				data[ k ] = data[ k ] + static_cast< T >( N );
-			}
-			if( i > j ) {
-				data[ i * N + j ] = data[ j * N + i ];
+				data[ j * N + i ] += static_cast< T >( N );
 			}
-
 		}
 	}
+	return data;
 }
 
+/** gnerate symmetric-hermitian positive definite matrix in a square container */
+template<
+	typename T
+>
+std::vector< T > generate_symmherm_pos_def_mat_data(
+	size_t N,
+	const typename std::enable_if<
+		grb::utils::is_complex< T >::value,
+		void
+	>::type * const = nullptr
+) {
+	return( generate_symmherm_pos_def_mat_data_full< T >( N ) );
```

Comment:
I think the return statement does not need to be wrapped in ().

Reply:
fixed, should be also moved to utils and replaced in all tests